tests/pagination/tests.py

```diff
@@ -303,6 +426,20 @@ def test_get_page_hook(self):
         self.assertEqual(page2.previous_page_number(), 1)
         self.assertIsNone(page2.next_page_number())
 
+    async def test_get_page_hook_async(self):
+        """
+        A AsyncPaginator subclass can use the ``_aget_page`` hook to
+        return an alternative to the standard AsyncPage class.
+        """
+        eleven = "abcdefghijk"
+        paginator = AsyncValidAdjacentNumsPaginator(eleven, per_page=6)
+        page1 = await paginator.apage(1)
+        page2 = await paginator.apage(2)
+        self.assertIsNone(await page1.aprevious_page_number())
+        self.assertEqual(await page1.anext_page_number(), 2)
+        self.assertEqual(await page2.aprevious_page_number(), 1)
+        self.assertIsNone(await page2.anext_page_number())
```

Comment:
```suggestion
        with self.assertRaises(EmptyPage):
            await page1.aprevious_page_number()
        self.assertEqual(await page1.anext_page_number(), 2)
        self.assertEqual(await page2.aprevious_page_number(), 1)
        with self.assertRaises(EmptyPage):
            await page2.anext_page_number()
```
I think the boundaries should be raising empty page errors

Reply:
https://github.com/django/django/blob/7d08d63740a99873220f6d14f5af751be648aed3/tests/pagination/custom.py#L21

I think it is correct that the EmptyPage exception does not occur because the class(`AsyncValidAdjacentNumsPage`) is being tested. 🤔